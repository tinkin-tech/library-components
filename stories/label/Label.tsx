import * as React from 'react'

export interface LabelPropsInterface {
  referenceId?: string
  label: string
  required?: boolean
  error?: boolean
  disabled?: boolean
}

export const LabelComponent: React.FC<LabelPropsInterface> = (
  props: LabelPropsInterface
) => {
  const { referenceId, label, required, error } = props

  return (
    <label
      className={`label ${error ? 'label-error' : ''}`}
      htmlFor={referenceId}
      data-testid="label-component"
    >
      {`${label || ''}${required ? '*' : ''}`}&nbsp;
    </label>
  )
}
