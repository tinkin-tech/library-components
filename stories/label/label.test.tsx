import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { LabelComponent } from './Label'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

const defaultLabel = (
  <LabelComponent
    label="Test Label"
    required={false}
    referenceId={null}
    disabled={false}
    error={false}
  />
)

describe('Label component', () => {
  it('Should render in virtual DOM', () => {
    const { getByTestId } = render(defaultLabel)
    expect(getByTestId('label-component')).toBeInTheDocument()
  })

  it('Should shot the snapshot with all properties', () => {
    const component = renderer.create(defaultLabel)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  })
})
