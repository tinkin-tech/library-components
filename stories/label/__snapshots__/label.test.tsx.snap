// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component Should shot the snapshot with all properties 1`] = `
<label
  className="label "
  data-testid="label-component"
  htmlFor={null}
>
  Test Label
  Â 
</label>
`;
